x-pack/plugins/observability_solution/apm/public/components/routing/templates/apm_main_template/inventory_callout.tsx

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiLink, EuiToolTip, EuiButtonIcon } from '@elastic/eui';
+import { TechnicalPreviewBadge } from '@kbn/observability-shared-plugin/public';
+import React from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { ENTITY_INVENTORY_LOCATOR_ID } from '@kbn/observability-shared-plugin/common';
+import { SerializableRecord } from '@kbn/utility-types';
+import { useApmPluginContext } from '../../../../context/apm_plugin/use_apm_plugin_context';
+
+export interface InventoryCalloutProps {
+  onDissmiss: () => void;
+}
+
+export function InventoryCallout({ onDissmiss }: InventoryCalloutProps) {
+  const { share } = useApmPluginContext();
+  const inventoryLocator = share.url.locators.get<SerializableRecord>(ENTITY_INVENTORY_LOCATOR_ID);
```

Comment:
Should we return null in case `inventoryLocator` is not defined?
```suggestion
  const inventoryLocator = share.url.locators.get<SerializableRecord>(ENTITY_INVENTORY_LOCATOR_ID);
  
  if(inventoryLocator === undefined){
    return null;
  }
```

Reply:
Now that I added the locator in the `ObservabilitySharedPlugin` locators (as you suggested [here](https://github.com/elastic/kibana/pull/193997#discussion_r1775220812)) this should never be undefined, right?

Reply:
You can check the type either by pressing ⌘ Command (or Cmd) and hover over the variable or try to access any property of inventoryLocator. TS should warn you but I think it _can_ be undefined

Reply:
Yes, it indicates that it can't be undefined. I just wanted to clarify that by implementing the suggested change, this particular check would no longer be necessary.

Reply:
>  I think it _can_ be undefined

I'm sharing your suggestion here to ensure complete clarity for everyone in case there's any confusion.

|Before|After|
|-|-|
|<img width="663" alt="Screenshot 2024-09-26 at 11 42 36" src="https://github.com/user-attachments/assets/f6d3d1a3-d7a1-4773-907d-258448477b62">|<img width="593" alt="Screenshot 2024-09-26 at 11 43 06" src="https://github.com/user-attachments/assets/642ca61e-8e8f-4f28-830c-2f91b329e247">|